gcovr/tests/test_args.py

```diff
@@ -0,0 +1,108 @@
+import sys
+import unittest
+from contextlib import contextmanager
+
+from ..__main__ import main, version_str
+
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+class CaptureObject:
+    def __init__(self, out, err, exception):
+        self.out = out
+        self.err = err
+        self.exception = exception
+
+
+class StringIOBuffered(StringIO):
```

Comment:
Is this class needed? I think a `StringIO.getvalue()` call is sufficient in `capture()`, no need to overrride `StringIO.close()`.

Reply:
`StringIO.getvalue()` will raise an exception if `StringIO.close()` has been called. `StringIOBuffered` offers `StringIOBuffered.data()` which will not raise an exception if `close()` has been called.